crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -1191,11 +1181,7 @@ where
                     })?
             };
 
-            ForkChoiceStrategy::Custom(match total_weight.cmp(&last_best_weight) {
-                Ordering::Greater => true,
-                Ordering::Equal => block_number > last_best_number,
-                Ordering::Less => false,
-            })
+            ForkChoiceStrategy::Custom(total_weight > last_best_weight)
```

Comment:
I don't think recovery from deep network partition is possible at all due to archiver inability to go back in time at the moment. For short-term forks we prefer the first block we see to build on, just like Bitcoin, but if later another fork happens to grow faster, we'll switch to it.

I was thinking that we might prefer shorter forks instead (since they will have heavier blocks on average), but I suspect there might be some issues with that.

I have left a question for Dariia in https://github.com/subspace/subspace/issues/1505, but this PR should be an improvement over previous implementation either way. I'm fairly certain we'll do more tweaks to this later.